src/audio/volume/volume.c

```diff
@@ -157,14 +158,31 @@ static struct comp_dev *volume_new(struct sof_ipc_comp *comp)
 	schedule_task_init(&cd->volwork, SOF_SCHEDULE_LL, SOF_TASK_PRI_MED,
 			   vol_work, dev, 0, 0);
 
-	/* set the default volumes */
+	/* Set the default volumes. If IPC sets min_value or max_value to
+	 * not-zero, use it. Otherwise set to internal limits and notify
+	 * ramp step calculation about assumed range with the ange set to
+	 * zero.
+	 */
+	if (vol->min_value || vol->max_value) {
+		cd->vol_min = MAX(vol->min_value, VOL_MIN);
+		cd->vol_max = MIN(vol->max_value, VOL_MAX);
```

Comment:
Here I was thinking if exceeding min and max limits VOL_MIN and VOL_MAX should be an error or silently limit it. The same is done for volume request elsewhere in this code. The VOL_MAX is a very high gain so such situation is very unlikely. But since ramp will fail with illegal values this limiting or error issuing is needed.

Reply:
What about notifying the user via trace_error if requested man or max cannot be used. This way at least it's obvious in the logs.

Reply:
Yes, makes sense. I'll add the error traces.